athenahealth/appointment_status.go

```diff
@@ -0,0 +1,42 @@
+package athenahealth
+
+// AppointmentStatus is derived from https://docs.athenahealth.com/api/api-ref/appointment#Book-appointment
+// The athenaNet appointment status. There are several possible statuses.
+// x=cancelled
+// f=future (It can include appointments where were never checked in, even if the appointment date is in the past. It is up to a practice to cancel appointments as a no show when appropriate to do so.)
+// o=open
+// 2=checked in
+// 3=checked out
+// 4=charge entered (i.e. a past appointment)
+type AppointmentStatus string
+
+const (
+	AppointmentStatusCancelled     AppointmentStatus = "x"
+	AppointmentStatusChargeEntered AppointmentStatus = "4"
+	AppointmentStatusCheckedIn     AppointmentStatus = "2"
+	AppointmentStatusCheckedOut    AppointmentStatus = "3"
+	AppointmentStatusFuture        AppointmentStatus = "f"
+	AppointmentStatusOpen          AppointmentStatus = "o"
+)
+
+var appointmentStatuses = []AppointmentStatus{
+	AppointmentStatusCancelled,
+	AppointmentStatusChargeEntered,
+	AppointmentStatusCheckedIn,
+	AppointmentStatusCheckedOut,
+	AppointmentStatusFuture,
+	AppointmentStatusOpen,
+}
+
+func (as AppointmentStatus) String() string {
+	return string(as)
+}
+
+func (as AppointmentStatus) Valid() bool {
+	for _, currentAS := range appointmentStatuses {
```

Comment:
is there any reason to keep this as an array instead of a map as I had previously?

Reply:
I think the main drawback of the map in its previous form was the redundancy/duplication of the string value.
I could see the map being valuable in quickly identifying `AppointmentStatus` validity by using the boolean returned from indexing the map. We'd prefer that the map's value be empty structs.
```go
var appointmentStatusMap = map[AppointmentStatus]struct{}{
	AppointmentStatusCancelled: {},
	AppointmentStatusChargeEntered: {},
	AppointmentStatusCheckedIn: {},
	AppointmentStatusCheckedOut: {},
	AppointmentStatusFuture: {},
	AppointmentStatusOpen: {},
}

func (as AppointmentStatus) Valid() bool {
	_, ok := appointmentStatusMap[as]
	return ok
}
```

Reply:
regarding polluting the internal package global namespace, I'd say it's probably fine to enclose `appointmentStatusMap` inside `Valid` until a use case for moving it out arises.